File: Java/BubbleSort.java
import java.io.*;
import java.util.Scanner;

class Array{
    int size;
    int[] arr;
    Array(int size){
        this.size = size;
        this.arr = new int[size];
    }
    void input(){
        Scanner sc = new Scanner(System.in);
        System.out.println("Enter the elements of the array: ");
        for(int i = 0; i < size; i++){
            arr[i] = sc.nextInt();
        }
    }
    void display(){
        for(int i = 0; i <size; i++){
            System.out.print(arr[i] + "\t");
        }
        System.out.println();
    }
    void bubblesort(){
        for(int i = size-1;i>0;i--){
            for(int j = 0; j < i; j++){
                if(arr[j] > arr[j+1]){
                    int temp = arr[j+1];
                    arr[j+1] = arr[j];
                    arr[j] = temp;
                }
            }
        }
    }
}

class BubbleSort{
    public static void main(String args[]){
        Scanner xy = new Scanner(System.in);
        System.out.println("Enter the size of the array: ");
        int size = xy.nextInt();
        Array new_arr = new Array(size);
        new_arr.input();
        new_arr.bubblesort();
        System.out.println("The sorted array is: ");
        new_arr.display();
    }
}

//sample output
/*
Enter the size of the array: 
10
Enter the elements of the array: 
14
5
56
4
11
2
3
86
1
-34
The sorted array is: 
-34     1       2       3       4       5       11      14      56      86
*/

File: C_programs/bubblesort.c
#include <stdio.h>

int Display(int *list, int size)
{
	for(int i = 0 ; i < size; i++)
	{
		printf("%d\t",list[i]);
	}
	printf("\n");
}

int Bubble_Sort(int *list,int size)
{
	int temp;
	for(int i = size-1; i > 1; i--)
	{
		for(int j = 0; j < i;j++)
		{
			if(list[j] > list[j+1])
			{
				temp = list[j+1];
				list[j+1] = list[j];
				list[j] = temp;
				Display(list,size);
			}
		}
	}
}


int main()
{
	int size;
	printf("Enter the size of the array: ");
	scanf("%d",&size);
	int list[size];
	printf("Enter the elements of the array:\n");
	for(int i = 0; i < size; i++)
	{
		scanf("%d",&list[i]);
	}
	Bubble_Sort(list,size);
	printf("***Final Sorted List***\n");
	Display(list,size);
}
